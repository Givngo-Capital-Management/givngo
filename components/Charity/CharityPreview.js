import React from 'react';
import { Button, Image } from 'semantic-ui-react';
import './CharityPreview.scss';

export default function CharityPreview(props) {
  const { horizontal } = props;
  return (
    <div
      className={['charity-preview', horizontal ? 'horizontal' : null].join(
        ' ',
      )}
    >
      <div className="image-container">
        <Image src="http://via.placeholder.com/210x118" />
        <div className="time-label">
          <span>05:22</span>
        </div>
      </div>

      <div className="charity-info">
        <div className="semi-bold show-max-two-lines">Video title</div>
        <div className="charity-preview-metadata-container">
          <div className="charity-title">Charity name</div>
          <div>
            <span>November 2, 2020</span>
            <p>Short description</p>
          </div>
        </div>
        <Button className="mini compact ui orange basic smaller">
          <i className="heart outline icon" />
          Donate
        </Button>
        <a href="www.google.ca" className="charity-preview-link">
          <i className="star outline icon" />
          Follow
        </a>
        <a href="www.google.ca" className="charity-preview-link">
          <i className="share icon" />
          Share
        </a>
      </div>
    </div>
  );
}
