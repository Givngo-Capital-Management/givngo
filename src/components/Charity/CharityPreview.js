import React from 'react';
import { Button, Image, Icon } from 'semantic-ui-react';
import styles from './CharityPreview.module.scss';
import moment from 'moment'
import Link from 'next/link';
import DonateModal from '../Stripe/DonateModal';

export default function CharityPreview(props) {
  const { horizontal, charity } = props;

  const id = charity ? charity.id : ""
  const name = charity ? charity.name : ""
  const videoDate = charity ? charity.video_date : ""
  const videoTitle = charity ? charity.video_title : ""
  const videoPreview = charity ? charity.video_preview : ""

  const date = moment.utc(videoDate * 1000).format('MMMM DD, YYYY').toString()

  const donateButton = (
    <Button className="mini compact ui orange basic smaller">
      <Icon name="heart outline" />
      Donate
    </Button>
  )

  return (
    <div
      className={[styles.preview, horizontal ? styles.horizontal : null].join(
        ' ',
      )}
    >
      <Link href={`/charity/${id}`}>
        <a>
          <div className={styles.image}>
            <Image src={videoPreview} />
            <div className={styles.time}>
              <span>05:22</span>
            </div>
          </div>
        </a>
      </Link>

      <div className={styles.info}>
        <Link href={`/charity/${id}`}>
          <a>
            <div className={`${styles.bold} ${styles.lines}`}>{videoTitle}</div>
          </a>
        </Link>
        <div className={styles.metadata}>
        <div className={styles.title}>{name}</div>
          <div>
            <span>{date}</span>
            <p>Short description</p>
          </div>
        </div>
        <DonateModal charity={charity} button={donateButton}/>
        <a href="www.google.ca" className={styles.link}>
          <Icon name="star outline" />
          Follow
        </a>
        <a href="www.google.ca" className={styles.link}>
          <Icon name="share" />
          Share
        </a>
      </div>
      
    </div>
  );
}
