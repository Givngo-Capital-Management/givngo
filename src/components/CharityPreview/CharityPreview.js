import React from 'react';
import './CharityPreview.scss';
import { Image } from 'semantic-ui-react';

export default function CharityPreview() {
  return (
    <div className="charity-preview">
      <div className="image-container">
        <Image src="http://via.placeholder.com/210x118" />
        <div className="time-label">
          <span>05:22</span>
        </div>

        <div className="charity-info">
          <div className="semi-bold show-max-two-lines">Video title</div>
          <div className="charity-preview-metadata-container">
            <div className="charity-title">Charity name</div>
            <div>
              <span>2 days ago</span>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}
