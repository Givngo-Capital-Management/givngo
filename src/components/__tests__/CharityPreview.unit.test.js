import React from 'react';
import { shallow } from 'enzyme';
import CharityPreview from '../Charity/CharityPreview';

describe('CharityPreview', () => {
  test('renders vertically', () => {
    const wrapper = shallow(<CharityPreview />);
    expect(wrapper).toMatchSnapshot();
  });
  test('renders horizontally', () => {
    const wrapper = shallow(<CharityPreview horizontal />);
    expect(wrapper).toMatchSnapshot();
  });
});
