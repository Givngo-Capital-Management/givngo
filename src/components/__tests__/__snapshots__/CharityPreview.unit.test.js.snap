// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharityPreview renders horizontally 1`] = `
<div
  className="preview horizontal"
>
  <Link
    href="/charity/"
  >
    <a>
      <div
        className="image"
      >
        <Image
          as="img"
          src=""
          ui={true}
        />
        <div
          className="time"
        >
          <span>
            05:22
          </span>
        </div>
      </div>
    </a>
  </Link>
  <div
    className="info"
  >
    <Link
      href="/charity/"
    >
      <a>
        <div
          className="bold lines"
        />
      </a>
    </Link>
    <div
      className="metadata"
    >
      <div
        className="title"
      />
      <div>
        <span>
          January 01, 1970
        </span>
        <p>
          Short description
        </p>
      </div>
    </div>
    <DonateModal
      button={
        <Button
          as="button"
          className="mini compact ui orange basic smaller"
        >
          <Icon
            as="i"
            name="heart outline"
          />
          Donate
        </Button>
      }
    />
    <a
      className="link"
      href="www.google.ca"
    >
      <Icon
        as="i"
        name="star outline"
      />
      Follow
    </a>
    <a
      className="link"
      href="www.google.ca"
    >
      <Icon
        as="i"
        name="share"
      />
      Share
    </a>
  </div>
</div>
`;

exports[`CharityPreview renders vertically 1`] = `
<div
  className="preview "
>
  <Link
    href="/charity/"
  >
    <a>
      <div
        className="image"
      >
        <Image
          as="img"
          src=""
          ui={true}
        />
        <div
          className="time"
        >
          <span>
            05:22
          </span>
        </div>
      </div>
    </a>
  </Link>
  <div
    className="info"
  >
    <Link
      href="/charity/"
    >
      <a>
        <div
          className="bold lines"
        />
      </a>
    </Link>
    <div
      className="metadata"
    >
      <div
        className="title"
      />
      <div>
        <span>
          January 01, 1970
        </span>
        <p>
          Short description
        </p>
      </div>
    </div>
    <DonateModal
      button={
        <Button
          as="button"
          className="mini compact ui orange basic smaller"
        >
          <Icon
            as="i"
            name="heart outline"
          />
          Donate
        </Button>
      }
    />
    <a
      className="link"
      href="www.google.ca"
    >
      <Icon
        as="i"
        name="star outline"
      />
      Follow
    </a>
    <a
      className="link"
      href="www.google.ca"
    >
      <Icon
        as="i"
        name="share"
      />
      Share
    </a>
  </div>
</div>
`;
